external-import/doppel/src/utils/config_helper.py

```diff
@@ -0,0 +1,126 @@
+import os
+
+import yaml
+from pycti import get_config_variable
+
+
+def load_config():
+    """Load YAML config file from the current directory and validate log level/type early."""
+    config_path = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "../config.yml"
+    )
+    config = {}
+
+    if os.path.isfile(config_path):
+        with open(config_path, "r", encoding="utf-8") as file:
+            config = yaml.safe_load(file) or {}
+
+    # Validate CONNECTOR_LOG_LEVEL
+    log_level = get_config_variable(
+        "CONNECTOR_LOG_LEVEL", ["connector", "log_level"], config, default="info"
+    ).lower()
+    valid_log_levels = ["info", "error", "debug", "warn"]
+    if log_level not in valid_log_levels:
+        raise ValueError(
+            f"Invalid CONNECTOR_LOG_LEVEL '{log_level}'. Allowed values: {valid_log_levels}"
+        )
+
+    # Validate CONNECTOR_TYPE
+    connector_type = get_config_variable(
+        "CONNECTOR_TYPE", ["connector", "type"], config
+    )
+    valid_types = [
+        "INTERNAL_ENRICHMENT",
+        "EXTERNAL_IMPORT",
+        "INTERNAL_IMPORT_FILE",
+        "INTERNAL_EXPORT_FILE",
+        "EXTERNAL_ENRICHMENT",
+        "STREAM",
+    ]
+    if not connector_type or connector_type not in valid_types:
+        raise ValueError(
+            f"Invalid CONNECTOR_TYPE '{connector_type}'. Allowed values: {valid_types}"
+        )
+
+    return config
+
+
+def validate_required_positive_integer(value, name):
+    if value is None or str(value).strip() == "":
+        raise ValueError(f"Missing required configuration: {name}")
+    try:
+        int_value = int(value)
+        if int_value <= 0:
+            raise ValueError(
+                f"Configuration '{name}' must be a positive integer. Got: {value}"
+            )
+    except ValueError as exc:
+        raise ValueError(
+            f"Configuration '{name}' must be a positive integer. Got: {value}"
+        ) from exc
+    return int_value
+
+
+def validate_required_string(value, name):
+    if not value:
+        raise ValueError(f"Missing required configuration: {name}")
+    return value
+
+
+def load_connector_config(config, helper):
+    try:
+        # Load raw values
+        api_key = get_config_variable("DOPPEL_API_KEY", ["doppel", "api_key"], config)
+        polling_interval = get_config_variable(
+            "POLLING_INTERVAL", ["doppel", "polling_interval"], config
+        )
+        max_retries = get_config_variable(
+            "MAX_RETRIES", ["doppel", "max_retries"], config
```

Comment:
As said above, the environment variable must match the yaml path:
DOPPEL_MAX_RETRIES
"doppel", "max_retries"
```suggestion
            "DOPPEL_MAX_RETRIES", ["doppel", "max_retries"], config
```